signer/migrations/0012__dkg_verification_extensions.sql

```diff
@@ -1,75 +1,52 @@
--- Enum table for DKG shares status
-CREATE TABLE sbtc_signer.dkg_shares_status (
-    -- The id of the status, not auto-incremented as we want to control the values.
-    id INTEGER PRIMARY KEY,
-    -- The name of the status.
-    key TEXT NOT NULL,
-    -- Brief description of what the status means.
-    description TEXT NOT NULL
-);
 
--- Insert the initial entries.
-INSERT INTO sbtc_signer.dkg_shares_status (id, key, description) VALUES
-    (0, 'PENDING', 'DKG round successful, pending verification via signing round'),
-    (1, 'VERIFIED', 'Successfully verified via signing round'),
-    (2, 'KEY_REVOKED', 'The DKG key has been revoked and should not be used');
+CREATE TYPE sbtc_signer.dkg_shares_status AS ENUM (
+    'unverified',
+    'verified',
+    'failed'
+);
 
 -- Add the new columns to the `dkg_shares` table. We're not adding indexes for
 -- now because the table is so small that the overhead likely outweighs the
 -- benefits.
 ALTER TABLE sbtc_signer.dkg_shares
-    -- Contains the current 
-    ADD COLUMN dkg_shares_status_id INTEGER DEFAULT 0 REFERENCES sbtc_signer.dkg_shares_status (id),
-    ADD COLUMN verified_at_bitcoin_block_hash BYTEA DEFAULT NULL,
-    ADD COLUMN verified_at_bitcoin_block_height BIGINT DEFAULT NULL,
-    ADD CONSTRAINT fk_dkg_shares_bitcoin_block_hash
```

Comment:
I think we should keep the FK -- if we don't have the bitcoin block when we're running DKG and we've _somehow_ gotten past the chain tip validations and are writing this, then something is wrong... I'm not sure we should be succeeding DKG in that state?

Reply:
I would be for not having it, eg suppose at some point we want to prune our bitcoin view but still keep the shares for whatever reason

Reply:
I'm fine either way -- tbh I'm not really sure of the purpose of having the block hash here at all:
- Wrt #1286, It's the block height that's mentioned as a sort of logical clock, I don't see a use for the block hash since the canonical height will never "go backwards".
- If it's for use as a salt for deterministic generation, I would probably just call this "salt" or something instead and treat it as such; then there's no implicit relation that may or may not exist.

Reply:
Since we think we'd want it for https://github.com/stacks-network/sbtc/issues/1303, I think it makes the most sense to keep it as-is for now. We can rename it though.